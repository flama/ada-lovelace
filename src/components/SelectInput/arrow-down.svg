<?xml version="1.0" encoding="UTF-8"?>
<svg width="9px" height="5px" viewBox="0 0 9 5" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 42 (36781) - http://www.bohemiancoding.com/sketch -->
    <title>Page 1</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Symbols" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Dropdpwn-/-Normal" transform="translate(-182.000000, -10.000000)" fill="#735D94">
            <g id="Page-1" transform="translate(182.000000, 10.000000)">
                <path d="M4.18125,3.48925 L0.83525,0.14325 C0.64425,-0.04775 0.33425,-0.04775 0.14325,0.14325 C-0.04775,0.33525 -0.04775,0.64425 0.14325,0.83525 L3.83525,4.52725 C4.02625,4.71825 4.33625,4.71825 4.52725,4.52725 L8.21925,0.83525 C8.31425,0.74025 8.36225,0.61525 8.36225,0.48925 C8.36225,0.36425 8.31425,0.23925 8.21925,0.14325 C8.02825,-0.04775 7.71825,-0.04775 7.52725,0.14325 L4.18125,3.48925 Z" id="Fill-1"></path>
            </g>
        </g>
    </g>
</svg>