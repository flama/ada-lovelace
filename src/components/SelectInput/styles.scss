@import '../../config';

.select-input {
  select {
    @include button;
    width: 200px;

    background-image: url('./arrow-down.svg?name=Arrow');
    background-repeat: no-repeat;
    background-position: 95% center;
  }
}
