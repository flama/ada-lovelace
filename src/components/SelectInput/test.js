import React from 'react'
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'

import SelectInput from '../SelectInput'

describe('SelectInput', () => {
  let selectInput
  let onChange
  let closeBubble

  beforeEach(() => {
    onChange = jest.fn()
    closeBubble = jest.fn()
    selectInput = mount(<SelectInput />)
  })

  it('renders correctly', () => {
    expect(toJSON(selectInput)).toMatchSnapshot()
  })

  it('calls callbacks when option changes', () => {
    selectInput.find('select').simulate('change')

    selectInput.setProps({
      onChange,
      closeBubble,
      options: [1, 2, 3]
    })

    expect(toJSON(selectInput)).toMatchSnapshot()

    selectInput.find('select').simulate('change')

    expect(onChange).toHaveBeenCalled()
    expect(closeBubble).toHaveBeenCalled()
  })
})
