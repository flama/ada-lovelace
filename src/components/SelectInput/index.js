import React from 'react'
import PropTypes from 'prop-types'

import './styles.scss'

const SelectInput = props => {
  let select

  let handleUserInput = () => {
    props.closeBubble()
    props.onChange(select.value)
  }

  let options = props.options.map(option => {
    return <option value={ option } key={ option }>{ option }</option>
  })

  return (
    <div className="select-input">
      <select required="required"
        onChange={ () => handleUserInput() }
        ref={ s => select = s }>
        <option value="-1">{ `Continents (All)` }</option>
        { options }
      </select>
    </div>
  )
}

SelectInput.defaultProps = {
  options: [],
  onChange: _=>_,
  closeBubble: _=>_
}

SelectInput.propTypes = {
  onChange: PropTypes.func,
  closeBubble: PropTypes.func,
  options: PropTypes.array
}

export default SelectInput
