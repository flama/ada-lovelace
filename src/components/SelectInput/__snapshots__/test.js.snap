// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput calls callbacks when option changes 1`] = `
<SelectInput
  closeBubble={[Function]}
  onChange={[Function]}
  options={
    Array [
      1,
      2,
      3,
    ]
  }
>
  <div
    className="select-input"
  >
    <select
      onChange={[Function]}
      required="required"
    >
      <option
        value="-1"
      >
        Continents (All)
      </option>
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
      <option
        value={3}
      >
        3
      </option>
    </select>
  </div>
</SelectInput>
`;

exports[`SelectInput renders correctly 1`] = `
<SelectInput
  closeBubble={[Function]}
  onChange={[Function]}
  options={Array []}
>
  <div
    className="select-input"
  >
    <select
      onChange={[Function]}
      required="required"
    >
      <option
        value="-1"
      >
        Continents (All)
      </option>
    </select>
  </div>
</SelectInput>
`;
